common/cluster/metadata.go

```diff
@@ -463,12 +467,14 @@ func (m *metadataImpl) refreshClusterMetadata(ctx context.Context) error {
 				InitialFailoverVersion: newClusterInfo.InitialFailoverVersion,
 				RPCAddress:             newClusterInfo.RPCAddress,
 				ShardCount:             newClusterInfo.ShardCount,
+				Tags:                   newClusterInfo.Tags,
 				version:                newClusterInfo.version,
 			}
 		} else if newClusterInfo.version > oldClusterInfo.version {
 			if newClusterInfo.Enabled == oldClusterInfo.Enabled &&
 				newClusterInfo.RPCAddress == oldClusterInfo.RPCAddress &&
-				newClusterInfo.InitialFailoverVersion == oldClusterInfo.InitialFailoverVersion {
+				newClusterInfo.InitialFailoverVersion == oldClusterInfo.InitialFailoverVersion &&
+				reflect.DeepEqual(newClusterInfo.Tags, oldClusterInfo.Tags) {
```

Comment:
[maps.Equal](https://pkg.go.dev/golang.org/x/exp/maps#Equal) should work in this case?